projects/igniteui-angular/src/lib/date-picker/date-picker.component.ts

```diff
@@ -886,7 +892,13 @@ export class IgxDatePickerComponent implements IDatePicker, ControlValueAccessor
                 this.hasHeader = false;
                 const dropDownOverlay =
                     (this.dropDownOverlaySettings !== undefined) ? this._dropDownOverlay : this._dropDownOverlaySettings;
-                dropDownOverlay.positionStrategy.settings.target = this.editableInputGroup.nativeElement;
+                if (this.editableInputGroup) {
+                    dropDownOverlay.positionStrategy.settings.target = this.editableInputGroup.nativeElement;
+                } else {
+                    // if the date picker is retemplated, set an element marked with #dropDownTarget as a target to the drop-down
+                    dropDownOverlay.positionStrategy.settings.target = this.templateDropDownTarget.nativeElement;
```

Comment:
If the user has retemplated the dropdown but didn't set the id `#dropDownTarget` to any element, this line will throw exception because `this.templateDropDownTarget` will be null.

Reply:
Added error message that #dropDownTarget should be set.